default/python/tests/AI/test_savegame_manager.py

```diff
@@ -56,35 +57,59 @@ class OldStyleClass():
     pass
 
 
-def test_simple_object_encoding():
-    def check_encoding(obj):
-        retval = savegame_codec.encode(obj)
-        assert retval
-        assert isinstance(retval, str)
+@fixture(
+    params=[
+        int(), float(), str(), bool(),
+        1, 1.2, 1.2e4, "TestString", True, False, None,
+        tuple(), set(), list(), dict(),
+        (1, 2, 3), [1, 2, 3], {1, 2, 3},
+        {'a': 1, True: False, (1, 2): {1, 2, 3}, (1, 2, (3, (4,))): [1, 2, (3, 4), {1, 2, 3}]}
+    ],
+    ids=str)
+def simple_object(request, ):
+    return request.param
+
+
+def check_encoding(obj):
+    retval = savegame_codec.encode(obj)
+    assert retval
+    assert isinstance(retval, str)
+
+    restored_obj = savegame_codec.decode(retval)
+    assert type(restored_obj) == type(obj)
+    assert restored_obj == obj
+
 
-        restored_obj = savegame_codec.decode(retval)
-        assert type(restored_obj) == type(obj)
-        assert restored_obj == obj
+def test_encoding_simple_object(simple_object):
+    check_encoding(simple_object)
 
-    for obj in (
-            int(), float(), str(), bool(),
-            1, 1.2, 1.2e4, "TestString", True, False, None,
-            tuple(), set(), list(), dict(),
-            (1, 2, 3), [1, 2, 3], {1, 2, 3},
-            {'a': 1, True: False, (1, 2): {1, 2, 3}, (1, 2, (3, (4,))): [1, 2, (3, 4), {1, 2, 3}]},
-            ):
-        check_encoding(obj)
 
-    for obj in (lambda x: 1, check_encoding, OldStyleClass, DummyTestClass):
-        with pytest.raises(savegame_codec.CanNotSaveGameException,
-                           message="Could save unsupported, potentially dangerous object."):
-            check_encoding(obj)
+def test_ecoding_function():
+    with pytest.raises(savegame_codec.CanNotSaveGameException,
+                       message="Could save function",
+                       match="Class __builtin__.function is not trusted"):
+        check_encoding(lambda: 0)
+
+
+def test_ecoding_old_style_class():
```

Comment:
Typo: encoding

Reply:
I usually go to previous message and add something like `, + check other cases`